web/src/filter.ts

```diff
@@ -248,7 +248,15 @@ export class Filter {
 
     constructor(terms?: Term[]) {
         if (terms === undefined) {
-            this._terms = [];
+            this._terms = [
+                // Set default filter to accept any message. This is different from
+                // `in:home` which means "messages that are in the home/`All messages` view".
+                {
+                    negated: false,
+                    operator: "in",
+                    operand: "all",
+                },
+            ];
```

Comment:
Hmm, I'm not sure I like this direction. I think my preference would be to just make `terms` a required argument? And then the home message list can pass its set of terms, and `all_messages_data` can pass an explicit `[]` set of filters, and `filter.ts` can be significantly cleaner.

Reply:
Makes sense. Done.